src/Lecture2.hs

```diff
@@ -101,7 +121,9 @@ spaces.
 
 🕯 HINT: look into Data.Char and Prelude modules for functions you may use.
 -}
-dropSpaces = error "TODO"
+
+dropSpaces :: [Char] -> [Char]
+dropSpaces list = filter ( \x -> x /= ' ') list
```

Comment:
This function hangs on infinite lists because `filter` walks through the entire list ♾️ 

Have a look at functions `dropWhile` and `takeWhile` to implement a lazier solution 🦥

Reply:
elegant solution achieved yaayy!! 😀 🎉